base/operators.jl

```diff
@@ -155,7 +155,7 @@ julia> "foo" ≠ "foo"
 false
 ```
 """
-!=(x, y) = !(x == y)::Bool
+!=(x, y) = !(x == y)
```

Comment:
I imagine there was a reason for the presence of this type assertion, so how problematic is it to remove it?

Reply:
https://github.com/JuliaLang/julia/pull/20792

Reply:
Best thing to do is try it and see what the impact on compiler performance is. Based on the motivation for #20792, it also might make sense just not to run inference on deprecated methods (that was already necessary for one method in #23273).